spec/models/ext_management_system_spec.rb

```diff
@@ -254,10 +254,23 @@
     end
   end
 
-  context "with virtual totals" do
+  context "with virtual totals and virtual aggregate" do
     before(:each) do
       @ems = FactoryGirl.create(:ems_vmware)
-      (1..2).each { |i| FactoryGirl.create(:vm_vmware, :ext_management_system => @ems, :name => "vm_#{i}") }
+      (1..2).each { FactoryGirl.create(:vm_vmware, :ext_management_system => @ems) }
+      (1..2).each do
+        FactoryGirl.create(:host,
```

Comment:
If I make this change the second host appear without hardware object
`FactoryGirl.create_list(:host, 4,
                              :ext_management_system => @ems,
                              :hardware              => FactoryGirl.create(:hardware, :cpu2x2, :ram1GB))`